### PR TITLE
Update RBAC permissions

### PR DESCRIPTION
**Pull request by yongja79**
- Update RBAC permissions to remove unnecessary permissions

Signed-off-by: Jason Yong <jason_yong@uk.ibm.com>

### CODE REVIEW

File: controllers/webspherelibertyapplication_controller.go [hunk 1]
```diff
@@ -51,15 +51,15 @@ const applicationFinalizer = "finalizer.webspherelibertyapps.liberty.websphere.i
...
+// +kubebuilder:rbac:groups=apps,resources=deployments;statefulsets,verbs=list;watch;create;update;delete,namespace=websphere-liberty-operator
 // +kubebuilder:rbac:groups=apps,resources=deployments/finalizers;statefulsets,verbs=update,namespace=websphere-liberty-operator
-// +kubebuilder:rbac:groups=core,resources=services;secrets;serviceaccounts;configmaps;persistentvolumeclaims,verbs=get;list;watch;create;update;patch;delete,namespace=websphere-liberty-operator
-// +kubebuilder:rbac:groups=autoscaling,resources=horizontalpodautoscalers,verbs=get;list;watch;create;update;patch;delete,namespace=websphere-liberty-operator
-// +kubebuilder:rbac:groups=networking.k8s.io,resources=ingresses,verbs=get;list;watch;create;update;patch;delete,namespace=websphere-liberty-operator
-// +kubebuilder:rbac:groups=route.openshift.io,resources=routes;routes/custom-host,verbs=get;list;watch;create;update;patch;delete,namespace=websphere-liberty-operator
+// +kubebuilder:rbac:groups=core,resources=services;secrets;serviceaccounts;configmaps;persistentvolumeclaims,verbs=list;watch;create;update;delete,namespace=websphere-liberty-operator
```
Comment (leochr) [inline controllers/webspherelibertyapplication_controller.go:56]: @yongja79 Thanks for the PR. 

`get` permission on Secret is needed for SSO (the relevant code is [here](https://github.com/WASdev/websphere-liberty-operator/blob/bc4cca50d4d01e72406e8ed56f412b6d000cf44d/utils/utils.go#L301).

So wondering why the SSO tests didn't fail even though the operator doesn't have permission to get Secret. 
Either these RBAC permissions are not used or the operator gets the permission via some other way. 

Deployment is the most used resource, so remove some/all obvious permissions for it to see if that results in any errors.

File: config/rbac/leader_election_role.yaml [hunk 2]
```diff
@@ -6,22 +6,29 @@ metadata:
 rules:
 - apiGroups:
   - ""
-  - coordination.k8s.io
   resources:
   - configmaps
-  - leases
   verbs:
+  - create
```
Comment (leochr) [inline config/rbac/leader_election_role.yaml:12]: This entire file is auto-generated by operator-sdk when operator was created. Not sure why it specified these permissions in the first place. Wondering if we should keep it as is, unless these wouldn't be needed for sure.

Comment (yongja79) [reply config/rbac/leader_election_role.yaml:12]: Removing those permissions doesn't seem to cause any errors or warnings in the log but there could be areas we are not hitting in our tests. I'll remove the changes in this file for now.

**Comment by yongja79** [comment]
@leochr added all the get permissions back so ready for re-review

File: bundle/manifests/ibm-websphere-liberty.clusterserviceversion.yaml [hunk 3]
```diff
@@ -613,9 +610,7 @@ spec:
           verbs:
           - create
           - delete
-          - get
```
Comment (leochr) [inline bundle/manifests/ibm-websphere-liberty.clusterserviceversion.yaml:616]: `make bundle` should be run to add this back I believe

**Review by leochr — COMMENTED** [review]
@yongja79 Thanks for the update. Just one comment about updating the bundle.

**Review by leochr — APPROVED** [review]
@yongja79 Looks good. Thanks.

**Pull request merged**